### PR TITLE
Gets Highlight Timestamps

### PR DESCRIPTION
**Pull request by ewega**
- Saves highlight timestamps as list of lists for specific URL
- Two results: timestamps for RF and timestamps for NN
- Currently hard-coded URL but different URL can be passed

Should I be sending the variables anywhere? Right now they're just sitting

### CODE REVIEW

File: post-processing/getHighlightTimestamps.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+# Get URL for which we want to do post-processing
+url = 'https://www.youtube.com/watch?v=-5NzaAt_wY0' # We would get this as a variable but hard-coded for now
+
+
+# Connect to DB and get appropriate information for URL
+db_conn_str = "mysql+pymysql://{:}:{:}@{:}/{:}".format(secrets.user, secrets.password, secrets.host, secrets.db)
+db_conn = create_engine(db_conn_str)
+df = pd.read_sql('SELECT * FROM predictions_table WHERE url="'+ url +'"', con=db_conn)
+
```
Comment (tjlam) [inline post-processing/getHighlightTimestamps.py:19]: add `db_conn.dispose()` to close the connection

File: post-processing/getHighlightTimestamps.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            highlight_detected_nn = True
+            highlight_start_nn = row['start']
+            highlight_end_nn = row['end']
+        else:
+            highlight_end_nn = row['end']
+    else:
+        if highlight_detected_nn == True: 
+            highlight_detected_nn = False
+            nn_timestamps.append([highlight_start_nn, highlight_end_nn])
+
```
Comment (tjlam) [inline post-processing/getHighlightTimestamps.py:51]: Bit of an edge case: If the last x rows are highlights then they wouldn't be added to the array.

File: post-processing/getHighlightTimestamps.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+# Call the the following variables for a list of list of highlight timestamps (depending on the model)
+rf_timestamps
+nn_timestamps
```
Comment (tjlam) [inline post-processing/getHighlightTimestamps.py]: Lets convert these arrays to a numpy array, and write them to a file named`videoId.npy`

**Review by tjlam — CHANGES_REQUESTED** [review]
Looks good just some small changes

File: post-processing/getHighlightTimestamps.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# Connect to DB and get appropriate information for URL
+db_conn_str = "mysql+pymysql://{:}:{:}@{:}/{:}".format(secrets.user, secrets.password, secrets.host, secrets.db)
+db_conn = create_engine(db_conn_str)
+df = pd.read_sql('SELECT * FROM predictions_table WHERE url="'+ url +'"', con=db_conn)
+
+# Loop through DFs and send highlight timestamps to arrays
+rf_timestamps, nn_timestamps = [], []
+highlight_detected_rf, highlight_detected_nn = False, False
+
+for index, row in df.iterrows():
```
Comment (mattlianje) [inline post-processing/getHighlightTimestamps.py]: `for row in df.iterrows():` should do it

File: post-processing/getHighlightTimestamps.py (hunk 3 above)
Comment (mattlianje) [reply post-processing/getHighlightTimestamps.py]: @Eldrick19 @tjlam good point - this got me thinking for moving on ... these highlights should be either in some db ...
or if we keep using npy files ... to an s3 bucket ... so tangentially related is this helper #57

**Review by mattlianje — COMMENTED** [review]
LGTM <3 just a small boi and Lam's stuff

File: post-processing/getHighlightTimestamps.py (hunk 1 above)
Comment (ewega) [reply post-processing/getHighlightTimestamps.py:19]: Good catch. Fixed in the latest push

File: post-processing/getHighlightTimestamps.py (hunk 2 above)
Comment (ewega) [reply post-processing/getHighlightTimestamps.py:51]: Good catch. Fixed in the latest push

File: post-processing/getHighlightTimestamps.py (hunk 3 above)
Comment (ewega) [reply post-processing/getHighlightTimestamps.py]: I agree. I saved them to `.npz` files for now (its like a `.npy` file but for more than one array). They are ID'd based off of the URL. Lemme know if we should identify them differently.

File: post-processing/getHighlightTimestamps.py (hunk 4 above)
Comment (ewega) [reply post-processing/getHighlightTimestamps.py]: With the new changes this make my code crash :/

**Review by tjlam — APPROVED** [review]
LGTM

File: post-processing/getHighlightTimestamps.py (hunk 4 above)
Comment (mattlianje) [reply post-processing/getHighlightTimestamps.py]: mmmm - nvm then) The new way you implemented is nice and more intuitive))

**Review by mattlianje — APPROVED** [review]
LGTM

**Pull request merged**